src/compositor/shells/mod.rs

```diff
@@ -10,6 +10,13 @@ pub enum Shell {
                                     * TODO Xdg */
 }
 
+impl Default for Shell {
+    fn default() -> Self {
+        // The default shell is always invalid
+        Shell::XdgV6(XdgV6ShellSurfaceHandle::new())
```

Comment:
The only awkward thing about this is I needed a `Default` for the view to avoid putting an ugly `Option` in the `ClientState`.

Reply:
Ah that's a shame. Not a big deal though, as long as we always set it in the client state